api/python/quilt3/packages.py

```diff
@@ -466,9 +470,9 @@ def resolve_hash(cls, registry, hash_prefix):
         return top_hash
 
     @classmethod
-    def shorten_tophash(cls, package_name, registry, top_hash):
+    def _shorten_tophash(cls, package_name, registry, top_hash):
```

Comment:
Why is this internal?

Reply:
I just wanted it to take the PhysicalKey, instead of doing all the config and URL parsing all over again.

Also, I'm not convinced that shorten_tophash is a good API to have: if collisions could realistically happen and there's a need to check for them, then how do you know a future hash won't cause a collision for the short hash you generate now?